### DISCUSSION TITLE
멀티 플레이어 테스트는 어떻게 하시나요?

### DISCUSSION THREAD

**Discussion by swcho** [naverz/zepeto-studio-kor#427] [5. Q&A]
이것 저것 혼자 삽질 중인데요.

여려명이 접속한 환경의 테스트는 어떻게 하시나요?
이왕이면 Unity 에디터 환경에서 단일 서버에 멀티로 띄울 수 있으면 좋겠는데 방법이 있는지 모르겠네요.

테스트 하실 때 어떻게 하시는지 궁금합니다.

감사합니다.

**Comment by pumfit** [comment]
유니티 에디터 환경에서 서버를 띄우고 테스트 하는 방식은 
동일한 프로젝트를 주고 받은 뒤 Custom Environment 를 만들어 접속하는 방식으로 진행하고 있습니다.

아래 가이드대로 진행하면 서버가 켜진 상태라면 여러명이서 동시에 접속이 가능합니다!
https://studio.zepeto.me/guides/accessing-to-multiplay-ip-addresses

**Reply by swcho** [reply]
감사합니다.
접근 가능한 주소가 아닌 것 같은데요.
혹시 댓글로 정리해 주시거나 스샷이라도 올려 주시면 감사 드리겠습니다. (--)(__)

**Reply by Zepeto-TechEvangelist** [reply]
가이드 주소 수정하였습니다 😄

**Reply by swcho** [reply]
@Zepeto-TechEvangelist @pumfit 
감사합니다~
가이드 정독을 먼저 해보도록 하겠습니다.

**Reply by swcho** [reply]
@Zepeto-TechEvangelist 
음. 결국 PC 두대로 각각 unity를 설치해서 해야 하는 걸까요?
로컬 빌드해서 애플리케이션을 만들어서 실행을해도 아에 local player character도 생성이 안된채로 나오네요.

**Reply by x4cf** [reply]
QR코드를 통한 모바일 테스트도 가능하답니다.
플레이어 캐릭터 생성과 연동은 가이드와 제페토 프로젝트 셈플을활용하시면 쉽게 구현하실 수 있어요!
https://studio.zepeto.me/guides/getting-started-with-templates

**Reply by swcho** [reply]
@x4cf 
네 답변 감사합니다.
모바일을 먼저 실행하고, Unity Run을 하면 두 명까지는 멀티 플레이 테스트가 가능한 것 같습니다.

MacOS를 사용 중 인데, desktop으로 빌드해서 실행해 보면 정상동작을 하지 않더군요.
이 것이 된다면, desktop 버전으로 여러개를 띄워 테스트 해 볼 수 있을 거 같은데 가능할지 모르겠습니다.

**Comment by swcho** [comment]
위에 답변 주신 것 처럼 개발 중에는 Mobile과 Unity IDE를 활용하여 총 2명의 멀티 플레이 테스트가 가능한 것으로 확인하였습니다.

나중에 더 좋은 방법이나 기능 추가가 있었으면 좋겠네요.

**Reply by smilejsu82** [reply]
에디터 2개에서는 불가능 한가요?~

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요. 저희가 제공하는 것은 아니지만 
유니티 에디터 2개를 띄워서 멀티플레이 테스트를 할 수 있는 ParrelSync 추천 드립니다.
https://github.com/VeriorPies/ParrelSync

**Reply by smilejsu82** [reply]
에디터 1 에서 서버 시작후 클라이언트 접속 
![image](https://user-images.githubusercontent.com/50188795/161470069-b37725b4-747d-4936-acda-b4b231dd5e6f.png)
에디터 2에서 커스텀 환경 셋팅 만들어서 포트 설정후 접속 
![image](https://user-images.githubusercontent.com/50188795/161470094-bce96d49-a47f-4e3a-a2ba-efa2e93b01a5.png)
![image](https://user-images.githubusercontent.com/50188795/161470112-91133b82-c4a5-46cd-b563-344649d6cd19.png)

이렇게는 안되나요?

**Reply by smilejsu82** [reply]
에디터1에서 서버 실행 
![image](https://user-images.githubusercontent.com/50188795/161471369-f450d54c-815d-4e1f-8079-8d43a50ee149.png)
![image](https://user-images.githubusercontent.com/50188795/161471385-bacd9116-1fd8-45a5-bc39-2d02bb266038.png)

에디터2에서 커스텀 환경설정 만들어 IP, Port 설정 
![image](https://user-images.githubusercontent.com/50188795/161471430-b90a3866-22b1-429b-b1c6-88e8426163fe.png)

ParrelSync설치후 
에디터2를 클론하고 열기 
![image](https://user-images.githubusercontent.com/50188795/161471469-7fb11c93-e30d-41b8-b493-ba77ab120bcc.png)


에디터2와 에디터2(클론) 실행결과 
세션이 동일하다고 나옵니다 
`Already added userid [61db41bf8d897e9a2f61e37b] -> exist sessionId : USER_hGOhSQJb7xX`

![image](https://user-images.githubusercontent.com/50188795/161471561-b90ccbe0-522b-4941-b543-7ebac13cb17c.png)

![image](https://user-images.githubusercontent.com/50188795/161472030-cb2fbcce-706a-4e5e-90eb-e109f2f4fb3d.png)


멀티플레이 개발시 에디터2개에서 테스트 해가면서 개발 하고싶은데 

무엇을 잘못 했을까요...?

**Reply by smilejsu82** [reply]
아이디를 다르게 접속 했더니  ParrelSync를 사용하지 않고도 잘 접속 되었습니다 

![image](https://user-images.githubusercontent.com/50188795/161472343-47386898-3364-4ba7-b1c3-7be371763111.png)


![image](https://user-images.githubusercontent.com/50188795/161472329-722a2ee2-435f-4826-ae79-8940bcdbd0b3.png)


![image](https://user-images.githubusercontent.com/50188795/161472320-d0281ade-2f22-4747-a9e8-a8888e3daecb.png)
